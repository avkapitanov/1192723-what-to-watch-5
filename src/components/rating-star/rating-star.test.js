import React from "react";
import renderer from "react-test-renderer";
import RatingStar from "./rating-star";

const noop = () => {};

describe(`RatingStar component render correctly`, () => {
  it(`Disabled, max 5`, () => {
    const tree = renderer.create(
        <RatingStar
          isDisabled={true}
          maxRating={5}
          onChangeHandler={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Enabled, max 5`, () => {
    const tree = renderer.create(
        <RatingStar
          isDisabled={false}
          maxRating={5}
          onChangeHandler={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});


