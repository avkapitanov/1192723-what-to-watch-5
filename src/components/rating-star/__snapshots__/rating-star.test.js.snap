// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingStar component render correctly Disabled, max 5 1`] = `
<div
  className="rating"
>
  <div
    className="rating__stars"
  >
    <input
      className="rating__input"
      disabled={true}
      id="star-1"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={1}
    />
    <label
      className="rating__label"
      htmlFor="star-1"
    >
      Rating 
      1
    </label>
    <input
      className="rating__input"
      disabled={true}
      id="star-2"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={2}
    />
    <label
      className="rating__label"
      htmlFor="star-2"
    >
      Rating 
      2
    </label>
    <input
      className="rating__input"
      disabled={true}
      id="star-3"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={3}
    />
    <label
      className="rating__label"
      htmlFor="star-3"
    >
      Rating 
      3
    </label>
    <input
      className="rating__input"
      disabled={true}
      id="star-4"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={4}
    />
    <label
      className="rating__label"
      htmlFor="star-4"
    >
      Rating 
      4
    </label>
    <input
      className="rating__input"
      disabled={true}
      id="star-5"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={5}
    />
    <label
      className="rating__label"
      htmlFor="star-5"
    >
      Rating 
      5
    </label>
  </div>
</div>
`;

exports[`RatingStar component render correctly Enabled, max 5 1`] = `
<div
  className="rating"
>
  <div
    className="rating__stars"
  >
    <input
      className="rating__input"
      disabled={false}
      id="star-1"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={1}
    />
    <label
      className="rating__label"
      htmlFor="star-1"
    >
      Rating 
      1
    </label>
    <input
      className="rating__input"
      disabled={false}
      id="star-2"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={2}
    />
    <label
      className="rating__label"
      htmlFor="star-2"
    >
      Rating 
      2
    </label>
    <input
      className="rating__input"
      disabled={false}
      id="star-3"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={3}
    />
    <label
      className="rating__label"
      htmlFor="star-3"
    >
      Rating 
      3
    </label>
    <input
      className="rating__input"
      disabled={false}
      id="star-4"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={4}
    />
    <label
      className="rating__label"
      htmlFor="star-4"
    >
      Rating 
      4
    </label>
    <input
      className="rating__input"
      disabled={false}
      id="star-5"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={5}
    />
    <label
      className="rating__label"
      htmlFor="star-5"
    >
      Rating 
      5
    </label>
  </div>
</div>
`;
